### DISCUSSION TITLE
ERRO jpeg: cannot open source image Filename.tif

### DISCUSSION THREAD

**Discussion by Sanookmakmak** [photoprism/photoprism#2651] [Q&A] [answered]
Error while indexing:
```
2022-08-27 14:23:19 ERRO jpeg: cannot open source image Filename.tif
2022-08-27 14:23:07 INFO convert: converting Filename.tif to Filename.tif.jpg (tiff)
```
Filename.tif has a size of about 83000x4000 pixels.
Therefore i assume that there must be a limit?

**Comment by graciousgrey** [comment]
You can configure the PHOTOPRISM_ORIGINALS_LIMIT --> https://docs.photoprism.app/getting-started/config-options/

**Reply by CDeLeon94** [reply]
I'm also having the same issue, the following setting did not resolve it:
```
PHOTOPRISM_ORIGINALS_LIMIT: -1
```

Appears to be an issue in the lzw compression of the files, however, the files open perfectly fine in all other applications. And these files were created with the same software as other successfully imported tifs (VueScan). 

```
2022-10-18 17:13:06 INFO moving files from import
2022-10-18 17:13:06 INFO import: found no .ppignore file
2022-10-18 17:13:06 INFO import: added folder /youth
2022-10-18 17:13:06 INFO metadata: 20221007_142634_8E1321F1.tif has an invalid local time offset (265h53m47s)
2022-10-18 17:13:06 INFO media: 20221007_142634_8E1321F1.tif was taken at 2022-10-07 14:26:34 +0000 UTC (meta)
2022-10-18 17:13:06 INFO import: moving main tiff file youth/20221007_142634_8E1321F1.tif to 2022/10/20221007_142634_A8248957.tif
2022-10-18 17:13:07 INFO convert: converting 20221007_142634_A8248957.tif to 20221007_142634_A8248957.tif.jpg (tiff)
2022-10-18 17:13:07 INFO metadata: 20221007_142634_8E1321F1.tif has an invalid local time offset (265h53m47s)
2022-10-18 17:13:07 ERRO jpeg: cannot open source image 20221007_142634_A8248957.tif
**2022-10-18 17:13:07 ERRO import: lzw: invalid code in 2022/10/20221007_142634_A8248957.tif (convert to jpeg)**
2022-10-18 17:13:09 INFO import completed in 2 s
```

**Comment by CorentinB** [comment]
I'm seeing the same issue here, any progress from anyone?

**Reply by lastzero** [reply]
JPEGs have a size limit of 30000 pixels, so a 83000x4000 TIFF could not be converted to JPEG (and would require a lot of memory). Also note the TIFF reader provided by Google as a part of the Go programming language does not support all TIFF features.

**Reply by CzechErface** [reply]
Hey lastzero!  I am an experienced dev and would like to contribute a fix for this if you could point me in the right direction.  I am sure that we can come up with a satisfactory solution.  Recently I have been scanning a lot of family albums at the maximum possible quality and about 80% of my images are encountering this error during processing.  Some of these TIFF files are upwards of 80MB each and have tons of pixels.

Please let me know if I can help in some way.

**Reply by lastzero** [reply]
Thanks for offering your help! From my understanding, this might only be a configuration issue if you are using our latest release:

https://docs.photoprism.app/getting-started/config-options/#storage

It includes improved TIFF support with a fallback to ImageMagick if the Go implementation can't handle it. It is also important that the image files have the correct filename extension, i.e. JPEG images should not end with .tif or vice versa.

**Reply by lastzero** [reply]
The related issue is this:
- #1612

Feel free to suggest improvements for future releases (e.g. with a pull request or a feature request).

**Reply by CzechErface** [reply]
Thanks for the quick reply.  I played around with the following ENV configurations options to no avail:

- PHOTOPRISM_JPEG_SIZE
- PHOTOPRISM_RESOLUTION_LIMIT
- PHOTOPRISM_ORIGINALS_LIMIT
- PHOTOPRISM_EXPERIMENTAL
- PHOTOPRISM_DISABLE_IMAGEMAGICK

I seem to have a special case on my hands because after some analysis I realized that images of nearly identical size *are* being processed just fine.  The logs do not indicate any other error than, e.g. `jpeg: cannot open source image photo_0427.tif` and all files are being created with the same software.

I am going to do a deeper dive into this when I have some time and see if I can locate the exact issue.  I will post an update here once I have solved it.

**Reply by CzechErface** [reply, marked as answer]
Well, as usual the issue turned out to be one of PBKAC.

First off, let me congratulate you guys on doing a good job maintaining this project.  The developer documentation was clear and concise and I was able to get a development build running easily.  Upon starting the development container and photoprism I copied over some of the problematic files and it worked without issue.  Finding this strange I decided to consult the `docker-compose.yml` that I am using for production and found that although it was on the `latest` revision, it was actually using an older image.

The solution:
```
docker-compose stop photoprism && docker-compose pull photoprism && docker-compose up -d
```

Once the *actual* latest image was pulled down, and the container re-created, everything worked perfectly!  Now that I have dipped my toe into the development waters I hope that I can contribute to real bugfixes and features moving forward.  I apologize for bothering you on the matter. 😅

Hopefully this post helps someone else that is mistakenly assuming that they are on the latest revision.
